src/serac/numerics/functional/dual.hpp

```diff
@@ -206,6 +206,48 @@ SERAC_HOST_DEVICE auto abs(dual<gradient_type> x)
   return (x.value >= 0) ? x : -x;
 }
 
+/** @brief implementation of max for dual numbers */
+template <typename gradient_type>
+SERAC_HOST_DEVICE auto max(dual<gradient_type> a, double b)
+{
+  return (a.value > b) ? a : b;
+}
+
+/** @brief implementation of max for dual numbers */
+template <typename gradient_type>
+SERAC_HOST_DEVICE auto max(double a, dual<gradient_type> b)
+{
+  return (a > b.value) ? a : b;
+}
```

Comment:
We should add tests for these mixed dual/double overloads. I don't think this implementation compiles-- the different branches of the ternary operator have different types.

Same for the dual/double `min` implementation.